static/app/components/charts/components/grid.tsx

```diff
@@ -11,7 +11,7 @@ export default function Grid(props: EChartOption.Grid = {}): EChartOption.Grid {
       top: 20,
       bottom: 20,
       // This should allow for sufficient space for Y-axis labels
-      left: '0%',
+      left: 4,
```

Comment:
This is necessary to prevent y-axis overflows. 

<img width="306" alt="Screen Shot 2021-09-27 at 12 46 44 PM" src="https://user-images.githubusercontent.com/44172267/134981082-b1d96e42-ccd3-4c5e-8f53-1e313b75a40a.png">

It looks like ECharts calculates the y-axis width by pre-rendering the axis and getting the pre-rendered width. This becomes a problem because there is no explicit way to tell ECharts to use tabular figures in axes (instead, we have to add a special CSS style to the chart container), so it will pre-render axes with the default proportional figures. Numbers with proportional figures are always more narrow than those with tabular figures. As a result, the calculated y-axis width is more narrow than it should be.

I'm not sure how we can cleanly fix this. The workaround I found is to add a left offset to the grid, thus providing more space for the y-axis. From my testing, an offset of 4 should be enough for most use cases.